### ISSUE TITLE
screens stop rendering until rejoin

### ISSUE THREAD

**Issue by shadowscion** [Rusketh/ExpAdv2#172]
Not _exactly_ sure how to reproduce this, it may have to do with rapidly placing and removing screens, but every once in a while screens will no longer render on your client until you rejoin. Restarting clientside code does nothing, nor does _expadv reload_.

**Comment by Rusketh** [comment]
Can you please tell me if this issue still exists.

**Comment by Orzlar** [comment]
This issue still exists.

**Comment by X-Coder** [comment]
I have seen it multiple times too, but a little different, very often my screens go completely green, the chip itself stays working fine, just the displays stops rendering. Reloading doesn't fix it.

Don't know how to reproduce it. But most of the time it happened right after a huge lag occurred on the server I was playing on. I know this green-screen issue from old wiremod times, it had the same issue on all their different screens on lag. They first fixed it by implementing a console command / button for reloading the screen rendering, but later they found another way to fix it. It was around the garrysmod 13 release. The holos were suffering from the same issue on lag and Nebual fixed it like this:

```lua
`	-- Severe lagspikes can detach the source entity from its lua, so we need to reapply things when its reattached
	hook.Add("NetworkEntityCreated", "wire_hologram_rescale", function(ent)
		if ent.scale and ent.DoScale then
			-- ent.scale isn't present on newly created holograms, only old ones that've been hit by a lagspike
			ent:DoScale()
			ent:DoClip()
			ent:DoVisible()
		end
	end)
```

**Comment by X-Coder** [comment]
Just found the old script wiremod team used for manually fixing broken screens because of lagspikes:
https://github.com/wiremod/wire/blob/master/lua/wire/client/rendertarget_fix.lua

They used the console command wire_rt_fix, but as said found later another way to fix it automatically.

**Comment by X-Coder** [comment]
It has gotten better, now the screens show the ea2 logo after lag and doesn't stay blank.
But it seems not to restore to state it was before the lag (at least if I use noFrameRefresh(true)).

Is there some "afterLag" event that gets called so I can manually reinitialize the ea2?

**Comment by shadowscion** [comment]
Perhaps the NetworkEntityCreated hook? It's what the holograms use to resync.

**Comment by X-Coder** [comment]
Yes NetworkEntityCreated is what Rusketh has used for fixing the screen rendering. I think the same hook could be used for rebinding the client side code back to the entitiy after the connection was lost during lag.

**Comment by Extra-Mental** [comment]
I've just done a test and it seems to still happen after a player initially spawns in, I tried with a bot, unfortunately I didn't have a spawn message to display but can confirm it happened when it spawned.
![4000_screenshots_20160407194513_1](https://cloud.githubusercontent.com/assets/13044464/14347405/52dc448a-fcfa-11e5-92fd-edc767df05ed.jpg)

**Comment by Ivef** [comment]
Is there a fix for this yet? I've noticed it will happen whenever a player spawns in, or if the server is hit with a lag spike

As far as I know the only way to fix this is in-game is to either change screen resolution ( if the above happens again, that resolution won't draw anymore ) or to re-join the server.

**Comment by Codingale** [comment]
It's still existant, it's exactly the same issue EGP had a while back IIRC. Divran might have hints. I've had this issue happen a few times recently.

I noticed the issue happens more if you start to desync or lag out. Additionally I've noticed my printColor clientside also fails along with a few other things if you lag out for a few seconds. It's caused me to have to replace my ExpAdv2 gates a few times in order to keep seeing the messages.
